Feature/firebase login operations

burakkoc5 opened the pull request:
Firebase authentication setup for signin, signup.
A login button that triggers the signin operation.
On success, the user is redirected to the home page.
On failure, an error message is displayed.
State management has been implemented to manage login states with Bloc.
Minor UI adjustments, such as button styling changes.

burakkoc5 committed:
firebase configure

burakkoc5 committed:
firebase config

burakkoc5 committed:
ios configs for firebase

burakkoc5 committed:
Fix: Prevent overflow issue on Registration Froms by adding scrollable view

burakkoc5 committed:
basic signin,signup, signout functionality added

burakkoc5 committed:
Added GetIt dependency for service locator pattern

burakkoc5 committed:
Implement state management with AuthCubit for Registration Process

- Integrated `AuthCubit` for handling authentication state in the signup and signin process.
- Display error messages from the authentication state in case of errors.
- Added password visibility toggle to the `ReTypePasswordTextfieldWidget`.

burakkoc5 committed:
translations added for all strings

burakkoc5 committed:
ripple effect added for app container button

burakkoc5 committed:
FirebaseAuthException extension created for error handling

E-MRE commented on android/app/google-services.json:
```diff
@@ -0,0 +1,29 @@
+{
```
gitignore ile bu dosyayı gizlemekte yarar var. dışarıdan görünmesi güvenlik açısından tehlike

E-MRE commented on firebase.json:
```diff
@@ -0,0 +1 @@
+{"flutter":{"platforms":{"android":{"default":{"projectId":"asset-tracker-3cc21","appId":"1:490137200342:android:52806d2e5cffbb33210b77","fileOutput":"android/app/google-services.json"}},"ios":{"default":{"projectId":"asset-tracker-3cc21","appId":"1:490137200342:ios:32fa08ba5eeb1236210b77","uploadDebugSymbols":false,"fileOutput":"ios/Runner/GoogleService-Info.plist"}},"macos":{"default":{"project ...
```
bunu da gizleyebiliriz.

E-MRE commented on lib/firebase_options.dart:
````diff
@@ -0,0 +1,88 @@
+///   options: DefaultFirebaseOptions.currentPlatform,
+/// );
+/// ```
+class DefaultFirebaseOptions {
````
bu dosya da gizlenebilir

E-MRE reviewed (changes requested):
Güzel olmuş, önemli dosyaları gizlersen çok daha iyi olacak

burakkoc5 committed:
gitingore updated

burakkoc5 committed:
Removed sensitive files from version control

burakkoc5 committed:
Removed Firebase configuration files from version control

E-MRE commented on lib/core/widgets/container_button.dart:
```diff
@@ -33,26 +33,29 @@ class AppContainerButton extends StatelessWidget {
+          borderRadius: Radiuses.sm.all,
+          child: Container(
+            padding: padding ??
+                EdgeInsets.only(left: isBig ? 16 : 8, right: isBig ? 16 : 4),
```
sizes

E-MRE commented on lib/main.dart:
```diff
@@ -1,13 +1,26 @@
-void main() {
+void main() async {
   WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
```
appInıt

E-MRE commented on lib/main.dart:
```diff
@@ -1,13 +1,26 @@
-  runApp(const MyApp());
+
+  runApp(MultiBlocProvider(providers: [
+    BlocProvider<AuthenticationCubit>(
```
multi provider

E-MRE reviewed (approved):
Eline sağlık

burakkoc5 committed:
Folder structure edited.
App intit and App provide classes added.

burakkoc5 commented:
### Merge Request: Authentication & Firebase Enhancements  

- **Firebase Integration:**  
  - Configured Firebase for Android/iOS.  
  - Added and later removed sensitive Firebase files from version control.  
  - Created `FirebaseAuthException` extension for error handling.  

- **Authentication Features:**  
  - Implemented sign-in, sign-up, and sign-out functionality.  
  - Integrated `AuthCubit` for state management during registration.  

- **UI Enhancements:**  
  - Fixed overflow issue on registration forms with a scrollable view.  
  - Added ripple effect for button interactions.  

- **Localization:**  
  - Added translations for all user-facing strings.  

- **Codebase Improvements:**  
  - Updated `.gitignore` to exclude sensitive files.  
  - Integrated `GetIt` for service locator pattern.  
  - Edited folder structure for better organization.  

- **Bug Fixes:**  
  - Addressed minor UI and functionality issues.  
  - Cleaned up and secured version control files.
